src/test/java/com/salesforce/dynamodbv2/testsupport/StreamsTestUtil.java

```diff
@@ -36,4 +42,30 @@ public static String mockSequenceNumber(int sequenceNumber) {
         return Arrays.stream(sequenceNumbers).mapToObj(StreamsTestUtil::mockRecord).collect(Collectors.toList());
     }
 
+    /**
+     * Verifies a cache miss/hit occurs on a describeStreamCache lookup.
+     */
+    public static DescribeStreamResult verifyDescribeStreamCacheResult(Cache describeStreamCache, String key,
+                                                                       boolean expectedCacheHit,
+                                                                       DescribeStreamResult expectedResult) {
+        DescribeStreamResult cacheLookupResult = (DescribeStreamResult) describeStreamCache.getIfPresent(key);
+
+        if (!expectedCacheHit) {
+            assertNull(cacheLookupResult);
+        } else {
+            System.out.println("cache lookup result: " + cacheLookupResult);
+            assertNotNull(cacheLookupResult);
+
+            assertTrue(cacheLookupResult.getStreamDescription().getStreamArn()
+                .equals(expectedResult.getStreamDescription().getStreamArn()));
+
+            if (expectedResult.getStreamDescription().getShards() != null) {
```

Comment:
Do you ever test the case where `expectedResult.getStreamDescription().getShards()` is null?

Reply:
added tests in f3d1d9ef8aec9f343ef093ee040b2fe75c8f169c